client/components/app.jsx

```diff
@@ -18,7 +21,34 @@ class App extends React.Component {
       .catch(error => console.error(error));
   }
 
+  addGrade(newGrade) {
+    fetch(
+      '/api/grades',
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(newGrade)
+      }
+    )
+      .then(res => res.json())
+      .then(grades => this.setState({ grades: this.state.grades.concat(newGrade) }))
+      .catch(error => console.error(error));
+  }
+
+  getAverageGrade() {
+    const { grades } = this.state;
+    this.avg = grades.reduce((accumulator, grade) => {
+      return accumulator + grade.grade;
+    }, 0);
+
+    this.avg = grades.length ? Math.ceil(this.avg / grades.length) : 0;
+  }
+
   render() {
+    this.getAverageGrade();
```

Comment:
In general, if your render method has function calls in it that don't return values, that should set off an alarm. This kind of method call is precisely a "side-effect". Instead, method calls in here should return values that can be passed as props to react elements.